monai/handlers/mlflow_handler.py

```diff
@@ -352,3 +402,46 @@ def _default_iteration_log(self, engine: Engine) -> None:
                     for i, param_group in enumerate(cur_optimizer.param_groups)
                 }
                 self._log_metrics(params, step=engine.state.iteration)
+
+    def _default_dataset_log(self, engine: Engine) -> None:
+        """
+        Execute dataset log operation based on MONAI `Workflow.data_loader.dataset` data.
+        Abstract sample names in a dataset and build a Pandas DataFrame from it. To use this
+        function, every sample in the input dataset must have a filename, which can be fetched
+        from the `filename_or_obj` parameter in the `image_meta_dict` of the sample.
+        This function will log a PandasDataset, generated from the Pandas DataFrame, to MLFlow
+        inputs. For more details about PandasDataset, please refer to this link:
+        https://mlflow.org/docs/latest/python_api/mlflow.data.html#mlflow.data.pandas_dataset.PandasDataset
+
+        Please note that it may take a while to record the dataset if it has too many samples.
+
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+
+        """
+        dataloader = getattr(engine, "data_loader", None)
+        dataset = getattr(dataloader, "dataset", None) if dataloader else None
+        if not dataset:
+            raise AttributeError("The dataset of the engine is None. Cannot record it by MLFlow.")
+
+        sample_dict: dict[str, list[str]] = {}
+        sample_dict["images"] = []
+        for sample in dataset:
```

Comment:
Will this line run transforms for all the images? Maybe it's very slow, you can add a TQDM progress bar?
@wyli What do you think?

Thanks.

Reply:
yes, if it's tracking a static dataset (doesn't change during workflow execution), perhaps this should be done before running the workflow, for example with a command `python -m monai.bundle log_dataset ...`

Reply:
`log_dataset` at first makes sense, but what if the dataset is too large if may also take a long time to do transforms on all the data? Seems here we only need to track the file name and number of the dataset here, actually, we don't need to do any transforms on them, right? Is it possible to use a JSON file here instead of looping the dataset? Seems we only set `dataset_dir` as `REQUIRED` in the bundle property.

Reply:
It would be better if there is a more efficient way to get all sample information without doing any transform on them.

Reply:
Can we use `dataset.data` here? However, this does not guarantee that the inputs are already loaded, but in this case, there is also no file_name.

Reply:
Thanks @KumoLiu , I think it's a more efficient way to get the sample information and will update this part accordingly.